UQ-PAC/bil-to-boogie-translator
# Performance tests

Change summary:

- Add timing to SystemTests
- Differentiate timeout from verification failure
- Have SystemTests write out two CSV files summarising the test results and times
- Pass CLI configuration through the program in runutils and the boogie translation
    - Add CLI flag `--boogie-use-lambda-stores` to generate the lambda definition for >8bit stores, rather than the nested MemOp definition, this gives reasonable performance without ArrayAxioms. 
- Add inline flag to BFunction

Breaking changes:
- boogie timeout is now always a test failure 

With defaults MemOps and ArrayAxioms:

```
[info] - summary
[info]   + Test summary: 305 verified, 81 failed to verify including 0 timeouts. 
[info]   + Average time to verify: 599 
[info]   + Average time to counterexample: 701
```

With lambdas and array theories:

```
[info] - summary
[info]   + Test summary: 305 verified, 81 failed to verify including 2 timeouts. 
[info]   + Average time to verify: 665 
[info]   + Average time to counterexample: 1851
```
